### ISSUE TITLE
Stack Overflow When Resizing SidePanel

### ISSUE THREAD

**Issue by geoffthemedio** [freeorion/freeorion#1086]
Start game, expand the population, resource, buildings, and military panels on the sidepanel. (Happened for me with just the bottom two open, but not with none open or just the bottom). Start resizing the sidepanel by dragging the bottom-right around. After a few seconds, FreeOrion crashes. Attached debugger indicates this is a stack overflow. Top of the stack is similar to:
````
>	GetResourceDir() Line 424	C++
 	ClientUI::ArtDir() Line 62	C++
 	AccordionPanel::SetCollapsed(bool collapsed=false) Line 94	C++
 	BuildingsPanel::DoLayout() Line 251	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	GG::Wnd::Resize(const GG::Pt & sz={...}) Line 538	C++
 	BuildingsPanel::DoLayout() Line 247	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	SidePanel::PlanetPanel::DoLayout() Line 963	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	AccordionPanel::SetCollapsed(bool collapsed=false) Line 108	C++
 	MilitaryPanel::DoLayout() Line 167	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	SidePanel::PlanetPanel::DoLayout() Line 958	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	AccordionPanel::SetCollapsed(bool collapsed=true) Line 108	C++
 	ResourcePanel::DoLayout() Line 196	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	SidePanel::PlanetPanel::DoLayout() Line 953	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	AccordionPanel::SetCollapsed(bool collapsed=true) Line 108	C++
 	PopulationPanel::DoLayout() Line 203	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	SidePanel::PlanetPanel::DoLayout() Line 948	C++
 	SidePanel::PlanetPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 1743	C++
 	SidePanel::PlanetPanelContainer::DoPanelsLayout() Line 2412	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	SidePanel::PlanetPanel::DoLayout() Line 978	C++
 	SidePanel::PlanetPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 1743	C++
 	GG::Wnd::Resize(const GG::Pt & sz={...}) Line 538	C++
 	SidePanel::PlanetPanel::DoLayout() Line 977	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	AccordionPanel::SetCollapsed(bool collapsed=false) Line 108	C++
 	BuildingsPanel::DoLayout() Line 251	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	GG::Wnd::Resize(const GG::Pt & sz={...}) Line 538	C++
 	BuildingsPanel::DoLayout() Line 247	C++
 	AccordionPanel::SizeMove(const GG::Pt & ul={...}, const GG::Pt & lr={...}) Line 89	C++
 	SidePanel::PlanetPanel::DoLayout() Line 963	C++
 	boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>::operator()<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(const boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > & connectionBody={...}) Line 120	C++
 	boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> >::dereference() Line 82	C++
 	boost::signals2::optional_last_value<void>::operator()<boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > >(boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > first={...}, boost::signals2::detail::slot_call_iterator_t<boost::signals2::detail::variadic_slot_invoker<boost::signals2::detail::void_type>,std::_List_iterator<std::_List_val<std::_List_simple_types<boost::shared_ptr<boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > > > >,boost::signals2::detail::connection_body<std::pair<enum boost::signals2::detail::slot_meta_group,boost::optional<int> >,boost::signals2::slot<void __cdecl(void),boost::function<void __cdecl(void)> >,boost::signals2::mutex> > last={...}) Line 58	C++
 	boost::signals2::detail::signal_impl<void __cdecl(void),boost::signals2::optional_last_value<void>,int,std::less<int>,boost::function<void __cdecl(void)>,boost::function<void __cdecl(boost::signals2::connection const &)>,boost::signals2::mutex>::operator()() Line 246	C++
 	AccordionPanel::SetCollapsed(bool collapsed=false) Line 108	C++
````

**Comment by geoffthemedio** [comment]
Last three commits (up to https://github.com/freeorion/freeorion/commit/eb4a976936fdb1ba5628175796d92bb3e8a22c45 ) stopped the issue for me, though I suspect there's some underlying problem with the layout logic.

**Comment by LGM-Doyle** [comment]
@geoffthemedio I did not see this bug when I sent the self-congratulatory message about you using `PreRender()`.  For a day I was happy `PreRender()` was being used.

I have duplicated this crash and will look into root causes.

**Comment by geoffthemedio** [comment]
To clarify, the issue is not dependent on the use of PreRender. Rather, using PreRender seems to avoid the issue occurring, though as noted, I suspect there is some lingering underlying cause with AccordionPanel layout.

**Comment by LGM-Doyle** [comment]
I think that you are correct that PreRender just makes things fast enough on our machines that the crash stops happening.

I think PR #1088 fixes the underlying problem.

**Comment by Vezzra** [comment]
> I think PR #1088 fixes the underlying problem.

That sounds as if you're not 100% sure, so I'd like to have a confirmation before I close this as resolved - @geoffthemedio, I've merged #1088, does that fix the issue for you?

**Comment by LGM-Doyle** [comment]
@Vezzra, I have done a backtrace of several crashes (on my machine) and am certain that commits two and three of #1088 fix two sources of infinite recursion that cause crashes when resizing the SidePanel when compiled with the commit that @geoffthemedio indicated.  I can not replicate the crash with this PR applied, even before the PreRender() speed up is applied. 

There is no clear (to me) update hierarchy for the SidePanel layout code, so there may be another potentially infinitely recursive path that was not triggered in the crash, did not show up in the backtraces and was not apparent to me in the code.

**Comment by geoffthemedio** [comment]
As noted in the PreRender patch, I couldn't replicate it before @LGM-Doyle (hopefully) fixed the underlying issue.

**Comment by Vezzra** [comment]
Ok, will close this issue. If there are other bugs buried in the code sections in question, we can open new issues for that.

**Comment by Vezzra** [comment]
Fixed by #1088.
